ReactiveCocoa/Swift/Flatten.swift

```diff
@@ -0,0 +1,552 @@
+//
+//  Flatten.swift
+//  ReactiveCocoa
+//
+//  Created by Neil Pankey on 11/30/15.
+//  Copyright © 2015 GitHub. All rights reserved.
+//
+
+/// Describes how multiple producers should be joined together.
+public enum FlattenStrategy: Equatable {
+	/// The producers should be merged, so that any value received on any of the
+	/// input producers will be forwarded immediately to the output producer.
+	///
+	/// The resulting producer will complete only when all inputs have completed.
+	case Merge
+
+	/// The producers should be concatenated, so that their values are sent in the
+	/// order of the producers themselves.
+	///
+	/// The resulting producer will complete only when all inputs have completed.
+	case Concat
+
+	/// Only the events from the latest input producer should be considered for
+	/// the output. Any producers received before that point will be disposed of.
+	///
+	/// The resulting producer will complete only when the producer-of-producers and
+	/// the latest producer has completed.
+	case Latest
+}
+
+
+extension SignalType where Value: SignalProducerType, Error == Value.Error {
+	/// Flattens the inner producers sent upon `signal` (into a single signal of
+	/// values), according to the semantics of the given strategy.
+	///
+	/// If `signal` or an active inner producer fails, the returned signal will
+	/// forward that failure immediately.
+	///
+	/// `Interrupted` events on inner producers will be treated like `Completed`
+	/// events on inner producers.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func flatten(strategy: FlattenStrategy) -> Signal<Value.Value, Error> {
+		switch strategy {
+		case .Merge:
+			return self.merge()
+
+		case .Concat:
+			return self.concat()
+
+		case .Latest:
+			return self.switchToLatest()
+		}
+	}
+}
+
+extension SignalProducerType where Value: SignalProducerType, Error == Value.Error {
+	/// Flattens the inner producers sent upon `signal` (into a single signal of
+	/// values), according to the semantics of the given strategy.
+	///
+	/// If `signal` or an active inner producer fails, the returned signal will
+	/// forward that failure immediately.
+	///
+	/// `Interrupted` events on inner producers will be treated like `Completed`
+	/// events on inner producers.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
```

Comment:
The comment should be about producer.
warn_unused_result should be "Did you forget to call `start` on the producer?"